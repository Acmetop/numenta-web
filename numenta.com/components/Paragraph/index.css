/* Numenta company website source code. Copyright Â© 2016 Numenta.*/
/* Full details in LICENSE.txt, or contact us at <http://numenta.com>.*/
/* This program is free software: you can redistribute it and/or modify it under*/
/* the terms of the GNU Affero General Public License as published by the Free*/
/* Software Foundation, either version 3 of the License, or (at your option) any*/
/* later version. This program is distributed in the hope that it will be*/
/* useful, but WITHOUT ANY WARRANTY; without even the implied warranty of*/
/* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero*/
/* General Public License for more details. You should have received a copy of*/
/* the GNU Affero General Public License along with this program. If not, see*/
/* <https://www.gnu.org/licenses/agpl.html>.*/

/* dropcap - sync with wrappers/md/md.css */
._dropcap:first-of-type::first-letter {
  background-color: #fff;
  color: #1aa0db;
  float: left;
  font-weight: 500;
  line-height: 0.85;
  margin: 0 0 -1rem 0;
  /* -s (<= mobile) */
  font-size: 6.8rem;
  padding: 0.5rem 0.3rem 0.3rem 0;
}
@media screen and (min-width: 48em) {
  ._dropcap:first-of-type::first-letter {
    /* -ns (> mobile) */
    font-size: 5.4rem;
    padding: 0.4rem 0.3rem 0.3rem 0;
  }
}


.paragraph {
  composes: lh-copy from 'tachyons';
  composes: measure-wide from 'tachyons';
  composes: measure-m from 'tachyons';
  composes: measure-wide-l from 'tachyons';
  composes: mt0 from 'tachyons';
}

.lead {
  composes: _dropcap;
}
