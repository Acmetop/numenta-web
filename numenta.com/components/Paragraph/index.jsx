import classNames from 'classnames'
import React from 'react'

import styles from './index.css'


const Paragraph = ({children, lead}) => {
  const classes = [styles.paragraph]

  if (lead === true) classes.push(styles.lead)

  return (
    <p className={classNames(...classes)}>
      {children}
    </p>
  )
}

Paragraph.propTypes = {
  children: React.PropTypes.any.isRequired,
  lead: React.PropTypes.bool.isRequired,
}

Paragraph.defaultProps = {
  lead: false,
}

export default Paragraph
