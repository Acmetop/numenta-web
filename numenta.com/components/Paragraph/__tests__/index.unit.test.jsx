import React from 'react'
import renderer from 'react-test-renderer'

import Paragraph from '../../Paragraph'


describe('<Paragraph /> React component unit test suite', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <Paragraph name="Jeff Hawkins" />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
