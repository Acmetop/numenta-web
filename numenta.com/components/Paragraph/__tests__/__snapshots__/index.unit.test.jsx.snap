exports[`<Paragraph /> React component unit test suite Renders correctly 1`] = `
<p
  className="paragraph" />
`;
